### ISSUE TITLE
Stale Icon Bug

### ISSUE THREAD

**Issue by erickcestari** [vercel/next.js#67665]
### Link to the code that reproduces this issue

https://github.com/erickcestari/Portfolio

## How to Reproduce

1. Clone the application.
2. Deploy it to Vercel.
3. Change the GitHub icon (update the image at the GitHub URL).
4. Clear the cache and rebuild the deployed application on Vercel.
5. Observe that the old GitHub icon still appears on the website.

## Code Example
```ts
import { ImageResponse } from 'next/og';

export const runtime = 'nodejs';

export const size = {
  width: 460,
  height: 460,
};
export const contentType = 'image/png';

export default function Icon() {
  const githubIconUrl = 'https://github.com/erickcestari.png';

  return new ImageResponse(
    (
      <img
        src={githubIconUrl}
        alt="GitHub Icon"
        style={{
          width: '100%',
          height: '100%',
        }}
      />
    ),
    {
      ...size,
    }
  );
}
```

I have changed my icon, but you can still see it's wrong at https://erickcestari.vercel.app/.

I tried rebuilding and clearing the cache, but it didn't work.

### Current vs. Expected behavior

# Current vs. Expected Behavior

## Current Behavior:

After changing the GitHub icon and deploying the application to Vercel, the old GitHub icon still appears on the website, even after clearing the cache and rebuilding the application.

## Expected Behavior:

After changing the GitHub icon and deploying the application to Vercel, the new GitHub icon should appear on the website immediately following a cache clear and rebuild.

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 10 Pro
  Available memory (MB): 16165
  Available CPU cores: 12
Binaries:
  Node: 21.1.0
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.2.3 // There is a newer version (14.2.5) available, upgrade recommended!
  eslint-config-next: 14.2.3
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.4.5
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Developer Experience

### Which stage(s) are affected? (Select all that apply)

Vercel (Deployed)

**Comment by huozhi** [comment]
This should be fixed on 14.2.5 with #66714 can you try the latest version and deploy again? Thanks 🙏

**Comment by erickcestari** [comment]
It worked for me. Thanks!
